### PR TITLE
DOC: linalg: fix typos in pinv docs

### PR DESCRIPTION
**Pull request by iamdipansh**
Fixed typo in documentation 
link of typo
https://github.com/scipy/scipy/blob/881b9fe85eae39015693ea3c83d406c252eebcab/scipy/linalg/_basic.py#L1385C55-L1385C55 https://github.com/scipy/scipy/blob/881b9fe85eae39015693ea3c83d406c252eebcab/scipy/linalg/_basic.py#L1415C9-L1415C9

Reference issue
DOC: typo in scipy.linalg.pinv.html #19063

What does this implement/fix?
Fixed the typo in documentation

### CODE REVIEW

File: scipy/linalg/_basic.py [hunk 1]
```diff
@@ -1414,7 +1414,7 @@ def pinv(a, *, atol=None, rtol=None, return_rank=False, check_finite=True,
     Here, ``A*`` denotes the conjugate transpose. The Moore-Penrose
     pseudoinverse is a unique ``B`` that satisfies all four of these
     conditions and exists for any ``A``. Note that, unlike the standard
-    matrix inverse, ``A`` does not have to be square or have
+    matrix inverse, ``A`` does not have to be square matrix and have linearly
```
Comment (j-bowhay) [inline scipy/linalg/_basic.py:1417]: Please revert the phrasing and just fix the typo

**Review by j-bowhay — CHANGES_REQUESTED** [review]
Thanks @iamdipansh, one comment.

Also since this is a doc only pr please add `[skip actions] [skip cirrus]` to save ci resources

File: scipy/linalg/_basic.py [hunk 2]
```diff
@@ -1414,7 +1414,7 @@ def pinv(a, *, atol=None, rtol=None, return_rank=False, check_finite=True,
...
     independant columns/rows.
```
Comment (j-bowhay) [inline scipy/linalg/_basic.py:1418]:
```suggestion
    matrix inverse, ``A`` does not have to be a square matrix or have
    linearly independent columns/rows.
```

**Review by j-bowhay — APPROVED** [review]
Thanks @iamdipansh, I made a small tweak but this looks good to go. Congratulations on your first contribution.

**Comment by iamdipansh** [comment]
Thanks @j-bowhay

**Pull request merged**